roles/ec2_instance_create/tasks/main.yml

```diff
@@ -0,0 +1,80 @@
+---
+- name: Run 'ec2_instance_create' role
+  module_defaults:
+    group/aws: "{{ aws_setup_credentials__output }}"
+  block:
+    - name: Create EC2 instance with provided configuration
+      amazon.aws.ec2_instance:
+        region: "{{ ec2_instance_create_aws_region }}"
+        name: "{{ ec2_instance_create_instance_name }}"
+        instance_type: "{{ ec2_instance_create_instance_type }}"
+        image_id: "{{ ec2_instance_create_ami_id }}"
+        key_name: "{{ ec2_instance_create_key_name }}"
+        vpc_subnet_id: "{{ ec2_instance_create_vpc_subnet_id }}"
+        tags: "{{ ec2_instance_create_tags | default(omit) }}"
+        wait: "{{ ec2_instance_create_wait_for_boot }}"
+      register: ec2_instance
+
+    - name: Create security group if enabled
+      when: ec2_instance_create_associate_external_sg is true
+      block:
+        - name: Define security group
```

Comment:
What if a security group with this name already exist?

Reply:
updated the name to be more unique in general. Also `ec2_instance_create_external_sg_name` param is provided for users to specify the sg name.
Additionally, added new param `ec2_instance_create_external_sg` which user can provide to associate instance to existing external sg.

Reply:
Hi @alinabuzachis, thank you for all your suggestions and they are all valid. This role has been added to enable automation experiences for the cloud content and has a hard deadline, with the specific use case already defined and documented. I'd like to add all these functionalities to the role in a separate PR. @mandar242, we can still include the delete functionality in this PR. If either of you could create an issue to track these points, it would be really helpful.

Reply:
@GomathiselviS @alinabuzachis 
sounds good. I'll keep track of these suggestions, and can work on adding those once this PR is merged.
For now, I'll work on adding the ability to delete instance and other resources created by the role.